experiment/evaluator.py

```diff
@@ -217,7 +292,33 @@ def _parse_libfuzzer_logs(self, log_handle) -> tuple[int, int, bool]:
         crashes = True
         continue
 
-    return cov_pcs, total_pcs, crashes
+    initcov, donecov, lastround = self._parse_fuzz_cov_info_from_libfuzzer_logs(
+        lines)
+
+    # NOTE: Crashes from incorrect drivers will not be counted.
+
+    if crashes:
+      # FP case 1: driver crashes at init or first few rounds.
+      if lastround is None or lastround <= EARLY_FUZZING_ROUND_THRESHOLD:
+        # No cov line has been identified or only INITED round has been passed.
+        # This is very likely the false positive cases.
+        return cov_pcs, total_pcs, True, True, 'FP_CRASH_NEAR_INIT'
+
+      # FP case 2: 1st func of the 1st thread stack is in driver.
+      crash_stacks = self._parse_stacks_from_libfuzzer_logs(lines)
+      for stack_frame in crash_stacks[:1]:
+        if self._stack_func_is_of_testing_project(stack_frame):
+          if 'LLVMFuzzerTestOneInput' in stack_frame:
+            return cov_pcs, total_pcs, True, True, 'FP_CRASH_IN_DRIVER'
+          break
+
+    else:
+      # Another error driver case: no cov increase.
```

Comment:
note: I believe we have seen cases where the fuzz target is legitimate even when there is no cov increase. This is often when a new target fuzzes a very buggy function that has never been fuzzed before, with a very shallow bug that is instantly triggered. 

@DonggeLiu to confirm if this is an issue. 

We may want to think about some kind of confidence score here instead of a binary yes/no for filtering crashes/fuzz targets.

Reply:
Sure, I will check with old known issues.

Reply:
BTW, would it be possible to capture `LeakSanitizer: detected memory leaks`?
They are usually not security-related and caused by the fuzz target, it would be useful to filter them (or give them a low confidence score). E.g.,
https://llm-exp.oss-fuzz.com/Result-reports/ofg-pr/2024-04-03-198-dg-comparison/sample/output-hiredis-rediscommand/01

Reply:
That's fantastic, much appreciated!